src/core/openthread.cpp

```diff
@@ -977,33 +979,33 @@ void HandleActiveScanResult(void *aContext, Mac::Frame *aFrame)
         memcpy(&result.mExtendedPanId, beacon->GetExtendedPanId(), sizeof(result.mExtendedPanId));
     }
 
-    handler(&result);
+    handler(&result, aClientContext);
 
 exit:
     return;
 }
 
 ThreadError otDiscover(uint32_t aScanChannels, uint16_t aScanDuration, uint16_t aPanId,
-                       otHandleActiveScanResult aCallback)
+                       otHandleActiveScanResult aCallback, void *aCallbackContext)
 {
     return sThreadNetif->GetMle().Discover(aScanChannels, aScanDuration, aPanId, &HandleMleDiscover,
-                                           reinterpret_cast<void *>(aCallback));
+                                           reinterpret_cast<void *>(aCallback), aCallbackContext);
```

Comment:
We need to save both the callback and callback context. Without allocating a new buffer in openthread.cpp to whole both (and pass a pointer to that down) the only other option is to have essentially a global callback context saved, which is exactly what I am trying to move away from.